python/lsst/pipe/base/pipeline.py

```diff
@@ -212,6 +222,13 @@ def addTask(self, task: Union[PipelineTask, str], label: str):
         else:
             raise ValueError("task must be either a child class of PipelineTask or a string containing"
                              " a fully qualified name to one")
+        if not label:
```

Comment:
Actually Thinking about this now, I dont think is possible with the new pipeline object to have a task that does not have a label specified.

Reply:
Hmm looking at the code in ctrl_mpexec for task with no label, I didnt remove the ? in the regex, and or change make pipeline to use the task name if there is no label. This will lead to weird/broken behavior. If you dont want to change on this ticket, that's fine. I can make a ticket and fix that behavior.

Reply:
I think we don't want to force users to always provide label on the command line so if label is missing then it should come from `_DefaultName`. To know `_DefaultName` I need to import task class and this is where I thought is the most natural place for it. I do not want to import anything when command line is parsed, and another potential place for that is in `CmdLineFwk` class but I think that if I do it here it will be more generic. Of course if you say that `Pipeline.addTask` method has to receive non-empty label then I'd simply add a check here and move `doImport` to `CmdLineFwk` instead.

Reply:
@natelust, let me know if you want me to move that `doImport` to `CmdLineFwk` before I merge both branches, should be easy for me to do, certainly faster than opening another ticket.

Reply:
I guess this is fine here, I really dont like the doImport in either place, as it is done again later. One thing we talked about doing in the future is not having a _DefaultName at all, and using the name of the task in places where _DefaultName would have been used. How would you feel about just using the string name of the class for the label here? I think @TallJimbo might have had and opinion as well.

Reply:
I'm OK with using class name instead of _DefaultName, but I know that _DefaultName has a long history and this should probably be discussed with wider audience. Just tell me what to do, I'll do it.

Reply:
Lets stick with `_DefaultName` for now.  I'd like to replace that with the unqualified Task name or something derived from it eventually (and then take advantage of that to e.g. avoid the `doImport` here), but until we've done that more globally, using the unqualified Task name here just exacerbates the problem of having too many names for a Task.

Reply:
OK, I'll merge it as it is now.